### ISSUE TITLE
no_inline_in_optional_branches: False positive - LOOP AT itab REFERENCE INTO DATA(x).

### ISSUE THREAD

**Issue by markstuppacher** [abaplint/abaplint#2522]
![image](https://user-images.githubusercontent.com/14176548/166247528-0677356a-9595-4e7c-8e0b-951290443c38.png)

**Comment by larshp** [comment]
yea, well, yea, well

if `itab` is empty, it can be argued that it is an optional branch?

I'm not really sure what the abap style guide considers optional

```abap
LOOP AT foo INTO DATA(foo1).
```

**Comment by markstuppacher** [comment]
The example in the styleguide uses the inline declared variable "outside" the optional branch. https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#dont-declare-inline-in-optional-branches

Maybe this shouldn't be reported:
`
LOOP AT foo INTO DATA(bar).
ENDLOOP.
`
And this should be reported:
`
LOOP AT foo INTO DATA(bar).
ENDLOOP.
bar = 'x'.
`
What do you think?

**Comment by larshp** [comment]
yea, that makes sense

but then the name "Don't declare inline in optional branches" is misleading, it should be "Dont use inline in unrelated branches" or something

**Comment by markstuppacher** [comment]
> but then the name "Don't declare inline in optional branches" is misleading
agree!

**Comment by larshp** [comment]
anyhow, for now, just disable the rule, it might be changed completely, or renamed/decommissioned as the style guide is misleading
